giraffe/src/utils/getTicks.ts

```diff
@@ -29,19 +26,23 @@ const getNumTicks = (
   charLength: number
 ): number => {
   const sampleTickWidth = sampleTick.length * charLength
-  const numTicks = Math.round((length / sampleTickWidth) * TICK_DENSITY)
-
-  return Math.max(numTicks, 2)
+  return Math.round(length / sampleTickWidth)
```

Comment:
it seems really easy for `sampleTickWidth` to become 0 - someone could pass in 0 to `charLength`, and then this would make some undefined behavior. if that's the case, we should definitely try to guard against that happening:

```typescript
// or find a better default than 0
return sampleTickWidth !== 0 ? Math.round(length / sampleTickWidth) : 0
```

Reply:
I'm not sure how the sampleTickWidth would become 0.  The `sampleTick` that is being based in is a timestamp:

https://github.com/influxdata/giraffe/pull/110/files#diff-325e130ef8529ac0832abade9a2f50bdR12

Reply:
@Asalem1
>I'm not sure how the `sampleTickWidth` would become 0. The `sampleTick` that is being based in is a timestamp:

`sampleTick` is all good, you're absolutely right! the issue is  `sampleTick` is [multiplied by `charLength`](https://github.com/influxdata/giraffe/blob/c20c75449edc745adc5555ff6fde2c195658e280/giraffe/src/utils/getTicks.ts#L28):
```ts
const sampleTickWidth = sampleTick.length * charLength
```
which is [passed into `getNumTicks`](https://github.com/influxdata/giraffe/blob/c20c75449edc745adc5555ff6fde2c195658e280/giraffe/src/utils/getTicks.ts#L13) 

```ts
const numTicks = getNumTicks(sampleTick, rangeLength, charLength)
```

by [outside input:](https://github.com/influxdata/giraffe/blob/c20c75449edc745adc5555ff6fde2c195658e280/giraffe/src/utils/getTicks.ts#L9)

```ts
export const getTicks = (
  domain: number[],
  rangeLength: number,
  charLength: number,
  formatter?: Formatter
): 
```
and so it's pretty untrustworthy at this level.